redis/redis.go

```diff
@@ -139,6 +140,56 @@ func (c *RedisStore[V]) SetEx(ctx context.Context, key string, value V, ttl time
 	return nil
 }
 
+func (c *RedisStore[V]) GetEx(ctx context.Context, key string) (V, time.Duration, bool, error) {
+	var out V
+	var ttl time.Duration
+
+	_, err := c.client.TxPipelined(ctx, func(pipe redis.Pipeliner) error {
+		getVal := pipe.Get(ctx, key)
+		getTTL := pipe.TTL(ctx, key)
+
+		_, err := pipe.Exec(ctx)
+		if errors.Is(err, redis.Nil) {
+			return err
+		}
+
+		if err != nil {
+			return fmt.Errorf("exec: %w", err)
+		}
+
+		ttl, err = getTTL.Result()
+		if err != nil {
+			return fmt.Errorf("TTL command failed: %w", err)
+		}
+
+		if ttl == -1 {
+			ttl = time.Duration(math.MaxInt64)
+		}
```

Comment:
btw: In here.. are you sure you're getting TTL time.Duration format from Redis? e.g. in nanoseconds?

Redis uses seconds afaik

Reply:
Redis has 2 commands for getting TTL: 

1. https://redis.io/docs/latest/commands/ttl/ - check how many seconds a given key will continue to be part of the dataset.
2. https://redis.io/docs/latest/commands/pttl/ - remaining time in seconds while PTTL returns it in milliseconds.